pkg/bspatch/bspatch.go

```diff
@@ -26,15 +26,23 @@
 package bspatch
 
 import (
+	"bufio"
 	"bytes"
 	"fmt"
 	"io"
 	"io/ioutil"
+	"os"
 
 	"github.com/dsnet/compress/bzip2"
 	"github.com/gabstv/go-bsdiff/pkg/util"
 )
 
+// Variables for streaming
+var (
+	WriteBufferSize = 4096
```

Comment:
This is way too small. I'd make this `512 * 1024` or even `1024 * 1024`, since it'll be at most half-utilized by the byte-adding logic if you take my suggestion below.

- using 1MB of extra memory is nothing, since Kite itself consumes hundreds of megabytes.
    * we could even go higher, but diminishing returns.
- it's probably optimal to make this similar to the L1 cache size, which can be 256/512K on modern processors.